app.py

```diff
@@ -480,16 +480,8 @@ def annotate(sentno):
 	senttok, _ = worker.postokenize(sent)
 	annotation, n = getannotation(username, id)
 	if annotation is not None:
-		if app.config['CGELVALIDATE'] is None:
-			item = exporttree(annotation.splitlines(), functions='add')
-			canonicalize(item.tree)
-			worker.domorph(item.tree)
-			tree = writediscbrackettree(item.tree, item.sent)
-		else: 
-			treeobj = ActivedopTree.from_str(annotation)
-			tree = treeobj.cgel_tree
 		return redirect(url_for(
-				'edit', sentno=sentno, annotated=1, tree=tree, n=n))
+				'edit', sentno=sentno, annotated=1, n=n))
```

Comment:
To check that I understand this change: `getannotation()` queries the database for the tree. This part of `annotate()` goes directly into edit mode if there is a saved tree. Instead of passing that tree in the request, we expect `edit()` to look it up. (So it is a redundant database lookup but maybe worth it to simplify the code.)

Reply:
Exactly -- this allows us to avoid passing the tree string as a URL parameter, but at the cost of an extra database query.